app/models/king.rb

```diff
@@ -6,4 +6,18 @@ def image
       '&#9818;'
     end
   end
+
+  def valid_move?(x,y)
+    if horizontal_move?(x_position, x, y_position, y)
```

Comment:
Looks good!  One suggestion that might make this method simpler - you can structure code in a style known as the "guard clause," which is where at the beginning of the method you make use of Ruby's predicate if style to return early if conditions aren't want you expect, "guarding" the rest of the method from executing.  An example would be:

```ruby
def valid_move?(x, y)
  return false if blocked?(x, y)
  # rest of the method
end
```

That way, if the piece would be blocked, it returns early and you don't have to keep checking `blocked?` multiple times.  You can even have more than one of them, so like:

```ruby
def valid_move?(x, y)
  return false if blocked?(x, y)
  return false if delta_y(y_position, y) > 1
  return false if delta_x(x_position, x) > 1
  # rest of the method
end
```

In fact, in this case, you can simply return `true` at the end of the method and the proceed it with guard clauses if you'd like - something like:

```ruby
def valid_move?(x, y)
  return false if blocked?(x, y)
  return false if guard_clause
  return false if another_guard_clause
  # ...

  true
end
```

Just an idea - Rubocop's also complaining about this code, but you can fix some of the issues there just by running `bundle exec rubocop -a` to activate it's auto-fix mode.